### ISSUE TITLE
Window Docking Bug in Latest Version of RStudio 2024.04.2 Build 764

### ISSUE THREAD

**Issue by kirito002** [rstudio/rstudio#14997]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2024.04.2 Build 764
    OS Version      : Windows 10 21H2 19044.1586
    R Version       : 

### Steps to reproduce the problem

1. Open RStudio 2024.04.2 Build 764 on Windows 10 21H2 19044.1586.
2. Open two windows and dock them to the left and right edges of the screen.
3. Use Ctrl+Enter to run code in one of the windows.

### Describe the problem in detail

In the new version of RStudio 2024.04.2 Build 764 on Windows 10 21H2 19044.1586, there is a window docking bug. When two windows are opened and automatically docked to the left and right edges, running code using Ctrl+Enter in one window causes the size and position of the other docked window to change, not maintaining the original docked position. This issue has been present since 2022, but no one has addressed or fixed it. I have previously used RStudio-2021.09.1-372 without encountering this problem. Detailed issue can be seen in the image below:
![微信图片_20240723111916](https://github.com/user-attachments/assets/9a91fb3d-610d-4b14-89bc-1caf2d319ebb)
![微信图片_20240723112033](https://github.com/user-attachments/assets/b42a2a0b-b002-4ca4-926f-f166fd8e3bab)

### Describe the behavior you expected

When running code using Ctrl+Enter in one docked window, the size and position of the other docked window should remain unchanged, maintaining the original docked position.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@kirito002 Thank you for raising the issue! Referring to older versions that you mention, is it possible that you first saw the problem in version 2022.07.0? That's when we shifted over to a different framework (Electron).

**Comment by kirito002** [comment]
> @kirito002 Thank you for raising the issue! Referring to older versions that you mention, is it possible that you first saw the problem in version 2022.07.0? That's when we shifted over to a different framework (Electron).
Yes I tried downloading the RStudio Desktop Electron version of RStudio-2022.07.0-548.exe (https://dailies.rstudio.com/version/2022.07.0+548.pro5/) and tried the above win10 paste attach ctrl+enter, no change in the position of the box on the right, it's good to have the paste attached. the newer version of 2024 is still having the problem, thank you very much!

**Comment by jbeaulie** [comment]
I have had the same issue for quite a while. Currently running:

RStudio 2024.09.1+394 "Cranberry Hibiscus" Release (a1fe401fc08c232d470278d1bc362d05d79753d9, 2024-11-03) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2024.09.1+394 Chrome/124.0.6367.243 Electron/30.4.0 Safari/537.36, Quarto 1.5.57

**Comment by jbeaulie** [comment]
The newest release seems to have resolved the problem:

RStudio
2025.05.0 Build 496
"Mariposa Orchid" Release (f0b76cc0, 2025-05-04) for windows
